### PR TITLE
Add JadooKB JKB65 RGB

### PR DESCRIPTION
**Pull request by wizard-gg**
## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

*  created new keyboard brand and add via keymap to keyboard

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/jadookb/jkb65/readme.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# JKB65 RGB
+
+
+‌<img src="https://jadookb.com/themes/custom-5125/JKB65.JPG" width="70%" height="70%" />
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/readme.md:4]: Please host this on imgur, and use Markdown image syntax.

File: keyboards/jadookb/jkb65/readme.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+* Hardware Supported: JKB65 RGB
+* Hardware Availability: [Jadoo Keyboard](https://jadookb.com/jkb65)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make jadookb/jkb65:default
+
+VIA example for this keyboard:
+
+    make jadookb/jkb65:default:via
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/readme.md:18]:
```suggestion
Flashing example for this keyboard:

    make jadookb/jkb65:default:flash
```

File: keyboards/jadookb/jkb65/rules.mk [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+LTO_ENABLE = yes
+TAP_DANCE_ENABLE = no
+LAYOUTS = 67_ansi
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/rules.mk:21]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = WS2812
LTO_ENABLE = yes
```

File: keyboards/jadookb/jkb65/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
+#include QMK_KEYBOARD_H
+
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    
+     [0] = LAYOUT_67_ansi(
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/keymaps/via/keymap.c:7]:
```suggestion
     [0] = LAYOUT(
```

File: keyboards/jadookb/jkb65/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_TILD,
+        KC_TAB,      KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,      KC_DEL,
+        KC_CAPS,       KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,              KC_PGUP,
+        KC_LSFT,            KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,          KC_PGDN,
+        KC_LCTL,   KC_LGUI,   KC_LALT,                       KC_SPC,                              MO(1),     KC_RCTL,   KC_LEFT,   KC_DOWN,    KC_RGHT
+    ),
+    [1] = LAYOUT_67_ansi(
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/keymaps/via/keymap.c:14]:
```suggestion
    [1] = LAYOUT(
```

File: keyboards/jadookb/jkb65/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______,          KC_PSCR,
+        _______,     RGB_RMOD, _______, RGB_MOD, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,      _______,
+        _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______,  _______, _______, _______,             _______,
+        _______,            _______,   _______,  _______, _______,  _______, _______, _______, _______, _______,  _______,  _______, RGB_VAI,        _______,
+        MAGIC_UNNO_GUI,MAGIC_NO_GUI,   _______,                      RGB_TOG,                              _______,   MO(2),   RGB_SPI,   RGB_VAD,    RGB_SPD
+    ),
+	[2] = LAYOUT_67_ansi(
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/keymaps/via/keymap.c:21]:
```suggestion
	[2] = LAYOUT(
```

File: keyboards/jadookb/jkb65/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        RESET, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______,     _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,      _______,
+        _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,
+        _______,            _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,        _______,
+        _______,   _______,   _______,                      _______,                              _______,   _______,   _______,   _______,    _______
+    ),
+	[3] = LAYOUT_67_ansi(
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/keymaps/via/keymap.c:28]:
```suggestion
	[3] = LAYOUT(
```

File: keyboards/jadookb/jkb65/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,36 @@
+#include QMK_KEYBOARD_H
+
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    
+     [0] = LAYOUT_67_ansi(
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/keymaps/default/keymap.c:7]:
```suggestion
     [0] = LAYOUT(
```

File: keyboards/jadookb/jkb65/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_TILD,
+        KC_TAB,      KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,      KC_DEL,
+        KC_CAPS,       KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,              KC_PGUP,
+        KC_LSFT,            KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,          KC_PGDN,
+        KC_LCTL,   KC_LGUI,   KC_LALT,                       KC_SPC,                              MO(1),     KC_RCTL,   KC_LEFT,   KC_DOWN,    KC_RGHT
+    ),
+    [1] = LAYOUT_67_ansi(
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/keymaps/default/keymap.c:14]:
```suggestion
    [1] = LAYOUT(
```

File: keyboards/jadookb/jkb65/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______,          KC_PSCR,
+        _______,     RGB_RMOD, _______, RGB_MOD, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,      _______,
+        _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______,  _______, _______, _______,             _______,
+        _______,            _______,   _______,  _______, _______,  _______, _______, _______, _______, _______,  _______,  _______, RGB_VAI,        _______,
+        MAGIC_UNNO_GUI,MAGIC_NO_GUI,   _______,                      RGB_TOG,                              _______,   MO(2),   RGB_SPI,   RGB_VAD,    RGB_SPD
+    ),
+	[2] = LAYOUT_67_ansi(
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/keymaps/default/keymap.c:21]:
```suggestion
	[2] = LAYOUT(
```

File: keyboards/jadookb/jkb65/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+        RESET, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______,     _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,      _______,
+        _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,
+        _______,            _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,        _______,
+        _______,   _______,   _______,                      _______,                              _______,   _______,   _______,   _______,    _______
+    ),
+	[3] = LAYOUT_67_ansi(
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/keymaps/default/keymap.c:28]:
```suggestion
	[3] = LAYOUT(
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by wizard-gg** [comment]
updated

**Review by wizard-gg — COMMENTED** [review]
full updated

File: keyboards/jadookb/jkb65/JKB65.c [hunk 12]
```diff
@@ -0,0 +1,84 @@
+#include "jkb65.h"
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/JKB65.c:1]: These two files still need to be lowercase.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/jadookb/jkb65/JKB65.c (hunk 12 above)
Comment (wizard-gg) [reply keyboards/jadookb/jkb65/JKB65.c]: done

File: keyboards/jadookb/jkb65/config.h [hunk 13]
```diff
@@ -0,0 +1,47 @@
+#pragma once
```
Comment (drashna) [inline keyboards/jadookb/jkb65/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/jadookb/jkb65/config.h [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+
+#define DIODE_DIRECTION COL2ROW
+
+#define RGB_DI_PIN F0
+
+#define DRIVER_LED_TOTAL 67
+#ifdef RGB_DI_PIN
+    #define RGBLED_NUM 67
+
+    #define RGB_DISABLE_WHEN_USB_SUSPENDED true
```
Comment (drashna) [inline keyboards/jadookb/jkb65/config.h:27]:
```suggestion
    #define RGB_DISABLE_WHEN_USB_SUSPENDED
```

File: keyboards/jadookb/jkb65/config.h [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+    #define RGBLIGHT_EFFECT_RAINBOW_MOOD
+    #define RGBLIGHT_SLEEP
+    #define RGB_DISABLE_TIMEOUT 0 
+    #define RGB_DISABLE_AFTER_TIMEOUT 0
+    #define RGBLIGHT_TIMEOUT 60000
+#endif
+
+#ifdef RGB_MATRIX_ENABLE
+    #define RGB_MATRIX_MAXIMUM_BRIGHTNESS 255
+#endif
```
Comment (drashna) [inline keyboards/jadookb/jkb65/config.h:39]: This is the default max, no need to set it. 
```suggestion
```

File: keyboards/jadookb/jkb65/jkb65.c [hunk 16]
```diff
@@ -0,0 +1,84 @@
+#include "jkb65.h"
```
Comment (drashna) [inline keyboards/jadookb/jkb65/jkb65.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/jadookb/jkb65/jkb65.h [hunk 17]
```diff
@@ -0,0 +1,18 @@
+#pragma once
```
Comment (drashna) [inline keyboards/jadookb/jkb65/jkb65.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/jadookb/jkb65/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,36 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/jadookb/jkb65/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/jadookb/jkb65/keymaps/via/config.h [hunk 19]
```diff
@@ -0,0 +1,4 @@
+#pragma once
```
Comment (drashna) [inline keyboards/jadookb/jkb65/keymaps/via/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/jadookb/jkb65/keymaps/via/keymap.c [hunk 20]
```diff
@@ -0,0 +1,36 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/jadookb/jkb65/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by wizard-gg — COMMENTED** [review]
updated

File: keyboards/jadookb/jkb65/jkb65.c [hunk 21]
```diff
@@ -0,0 +1,84 @@
...
+void check_rgb_timeout() {
+  if (!is_rgb_timeout && timer_elapsed(key_timer) > RGBLIGHT_TIMEOUT) {
+    rgblight_suspend();
+    is_rgb_timeout = true;
+  }
+}
+
+
+
+void housekeeping_task_user(void) {
```
Comment (drashna) [inline keyboards/jadookb/jkb65/jkb65.c:54]:
```suggestion
void housekeeping_task_kb(void) {
```

File: keyboards/jadookb/jkb65/jkb65.c [hunk 22]
```diff
@@ -0,0 +1,84 @@
...
+  #endif
+  
+}
+
+void post_process_record_user(uint16_t keycode, keyrecord_t *record) {
+  #ifdef RGBLIGHT_TIMEOUT
+  if (record->event.pressed) refresh_rgb();
+  #endif
+
+}
```
Comment (drashna) [inline keyboards/jadookb/jkb65/jkb65.c:66]:
```suggestion
void post_process_record_kb(uint16_t keycode, keyrecord_t *record) {
#ifdef RGBLIGHT_TIMEOUT
    if (record->event.pressed) { refresh_rgb(); }
#endif
    post_process_record_user(keycode, record);
}
```

File: keyboards/jadookb/jkb65/jkb65.c [hunk 23]
```diff
@@ -0,0 +1,84 @@
...
+
+void post_encoder_update_user(uint8_t index, bool clockwise) {
+  #ifdef RGBLIGHT_TIMEOUT
+  refresh_rgb();
+  #endif
+
+}
```
Comment (drashna) [inline keyboards/jadookb/jkb65/jkb65.c:73]:
```suggestion
bool encoder_update_kb(uint8_t index, bool clockwise) {
#ifdef RGBLIGHT_TIMEOUT
    refresh_rgb();
#endif

    return encoder_update_user(index, clockwise);
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by wizard-gg — COMMENTED** [review]
updated

File: keyboards/jadookb/jkb65/readme.md [hunk 24]
```diff
@@ -0,0 +1,23 @@
+# JKB65 RGB
+
+![JKB65RGB](https://i.imgur.com/CKtVyyI.jpg)
```
Comment (fauxpark) [inline keyboards/jadookb/jkb65/readme.md:3]:
```suggestion
![JKB65RGB](https://i.imgur.com/CKtVyyIh.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by wizard-gg — COMMENTED** [review]
updated

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**